python/tk_multi_loader/dialog.py

```diff
@@ -1095,164 +1097,177 @@ def _load_entity_presets(self):
+
+            if not type_hierarchy:
+
+                # Add a layout to host search.
+                search_layout = QtGui.QHBoxLayout()
+                layout.addLayout(search_layout)
+
+                # Add the search text field.
+                search = QtGui.QLineEdit(tab)
+                search.setStyleSheet("QLineEdit{ border-width: 1px; "
+                                                "background-image: url(:/res/search.png); "
+                                                "background-repeat: no-repeat; "
+                                                "background-position: center left; "
+                                                "border-radius: 5px; "
+                                                "padding-left:20px; "
+                                                "margin:4px; "
+                                                "height:22px; "
+                                                "}")
+                search.setToolTip("Use the <i>search</i> field to narrow down the items displayed in the tree above.")
+
+                try:
+                    # This was introduced in Qt 4.7, so try to use it if we can...
+                    search.setPlaceholderText("Search...")
+                except:
+                    pass
+
+                search_layout.addWidget(search)
+
+                # Add a cancel search button, disabled by default.
+                clear_search = QtGui.QToolButton(tab)
+                icon = QtGui.QIcon()
+                icon.addPixmap(QtGui.QPixmap(":/res/clear_search.png"), QtGui.QIcon.Normal, QtGui.QIcon.Off)
+                clear_search.setIcon(icon)
+                clear_search.setAutoRaise(True)
+                clear_search.clicked.connect(lambda editor=search: editor.setText(""))
+                clear_search.setToolTip("Click to clear your current search.")
+                search_layout.addWidget(clear_search)
+
+                # Drive the proxy model with the search text.
+                search.textChanged.connect(lambda text, v=view, pm=proxy_model: self._on_search_text_changed(text, v, pm))
+
+                # Keep a handle to all the new Qt objects, otherwise the GC may not work.
+                self._dynamic_widgets.extend([search_layout,
+                                              search,
+                                              clear_search,
+                                              icon])
+
+            # Set up a view right click menu.
             action_ea = QtGui.QAction("Expand All Folders", view)
-            action_ca = QtGui.QAction("Collapse All Folders", view)
-            action_refresh = QtGui.QAction("Refresh", view)
-
             action_ea.triggered.connect(view.expandAll)
-            action_ca.triggered.connect(view.collapseAll)
-            action_refresh.triggered.connect(model.async_refresh)
             view.addAction(action_ea)
+            action_ca = QtGui.QAction("Collapse All Folders", view)
```

Comment:
seems like expand all folders shouldn't be allowed for the hierarchy model - can we take it out?